org.scala-ide.sdt.aspects/src/scala/tools/eclipse/contribution/weaving/jdt/jcompiler/BuildManagerStore.java

```diff
@@ -81,42 +86,60 @@ public void setJavaSourceFilesToCompile(File[] files, IProject project) {
   }
 
   private IProject project(AbstractImageBuilder imageBuilder) {
-	Field[] fields = AbstractImageBuilder.class.getDeclaredFields();
-	Field javaBuilderField = null;
-	for (Field f: fields) {
-	  if ("javaBuilder".equals(f.getName())) {
-	    javaBuilderField = f;
-	    javaBuilderField.setAccessible(true);
-	  } 
-	}
-	JavaBuilder builder = null;
-	try {
-	  builder = (JavaBuilder)javaBuilderField.get(imageBuilder);
-	  if (builder == null) {
-	    throw new IllegalArgumentException("java builder of image builder is null");
-	  }
-	} catch(Exception e) {
-	  throw new IllegalArgumentException("image builder met problems with retrieving java builder", e);
-	}
+    Field[] fields = AbstractImageBuilder.class.getDeclaredFields();
+    Field javaBuilderField = null;
+    for (Field f : fields) {
+      if ("javaBuilder".equals(f.getName())) {
+        javaBuilderField = f;
+        javaBuilderField.setAccessible(true);
+      }
+    }
+    JavaBuilder builder = null;
+    try {
+      builder = (JavaBuilder) javaBuilderField.get(imageBuilder);
+      if (builder == null) {
+        throw new IllegalArgumentException(
+            "java builder of image builder is null");
+      }
+    } catch (Exception e) {
+      throw new IllegalArgumentException(
+          "image builder met problems with retrieving java builder", e);
+    }
     return builder.getProject();
   }
 
   /**
-   * Defensively filters <code>SourceFile</code> of current project in given scope.<br/>
+   * Defensively filters <code>SourceFile</code> of current project in given
+   * scope.<br/>
+   * 
    * @param sources
    * @param imageBuilder
-   * @return these <code>SourceFile</code> which resources belong to current compilation scope
+   * @return these <code>SourceFile</code> which resources belong to current
+   *         compilation scope
    */
   public List<SourceFile> filterProjectSources(List<SourceFile> sources, AbstractImageBuilder imageBuilder) {
-	List<SourceFile> sourcesToCompile = new ArrayList<SourceFile>();
-	File[] scopeProjectSources = projectToJavaSourceFiles.get(project(imageBuilder));
-	List<File> projectFiles = scopeProjectSources != null ? asList(scopeProjectSources) : new ArrayList<File>();
-	for (SourceFile source: sources) {
-	  File file = source.resource.getRawLocation().makeAbsolute().toFile();
-	  if (projectFiles.contains(file)) {
-		sourcesToCompile.add(source);
-	  }
-	}
-	return sourcesToCompile;
+    IProject project = project(imageBuilder);
+    if (!isScalaProject(project)) {
+      return new ArrayList<SourceFile>(sources);
```

Comment:
Minor thing, but do we need to copy `sources` into a new array list? Couldn't we simply return `sources` here?